apps/cowswap-frontend/src/modules/trade/hooks/useAmountsToSign.ts

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from 'react'
+
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { useDerivedTradeState } from './useDerivedTradeState'
+import { useReceiveAmountInfo } from './useReceiveAmountInfo'
+
+export interface AmountsToSign {
+  maximumSendSellAmount: CurrencyAmount<Currency>
+  minimumReceiveBuyAmount: CurrencyAmount<Currency>
+}
+
+/**
+ * Returns amounts that will end up in signed order
+ * It means, those values already include fees and slippage depending on order type
+ */
+export function useAmountsToSign(): AmountsToSign | null {
```

Comment:
All the logic is concentrated here.
For widgets which don't support buy orders we can't rely on `ReceiveAmountInfo` because this data relies on `tradeQuotesAtom`.
In Limit Orders and TWAP we use quotes, but we don't really use quote results in the final order because user can specify the price manually.
Similar in TWAP, we use quote, but only for a part amount. The whole buy amount is calculated as partAmountAfterSlippage * numOfParts.
So, we use `afterSlippage` from `useReceiveAmountInfo()` only for Swap, because only there we use quote results as it is depending on order type.